consumer/src/test/java/no/nav/sbl/dialogarena/modiabrukerdialog/consumer/service/oppgavebehandling/RestOppgaveBehandlingServiceImplTest.kt

```diff
@@ -0,0 +1,607 @@
+import no.nav.sbl.dialogarena.abac.AbacResponse
+import no.nav.sbl.dialogarena.abac.Decision
+import no.nav.sbl.dialogarena.abac.Response
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.Temagruppe
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.oppgave.generated.apis.OppgaveApi
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.oppgave.generated.models.*
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.pdl.generated.HentIdent
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.LeggTilbakeOppgaveRequest
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.OpprettOppgaveRequest
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.RestOppgaveBehandlingService
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.arbeidsfordeling.ArbeidsfordelingV1Service
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.norg.AnsattService
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.service.pdl.PdlOppslagService
+import no.nav.sbl.dialogarena.modiabrukerdialog.api.utils.http.SubjectHandlerUtil
+import no.nav.sbl.dialogarena.modiabrukerdialog.consumer.service.kodeverksmapper.KodeverksmapperService
+import no.nav.sbl.dialogarena.modiabrukerdialog.tilgangskontroll.Tilgangskontroll
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import org.junit.jupiter.api.Test
+import java.time.LocalDate.now
+import java.util.*
+
+class RestOppgaveBehandlingServiceImplTest {
+    val apiClient: OppgaveApi = mockk()
+    val kodeverksmapperService: KodeverksmapperService = mockk()
+    val pdlOppslagService: PdlOppslagService = mockk()
+    val tilgangskontroll: Tilgangskontroll = Tilgangskontroll(RestOppgaveMockFactory.tilgangskontrollContext)
+    val ansattService: AnsattService = mockk()
+    val arbeidsfordelingService: ArbeidsfordelingV1Service = mockk()
+    val stsService: SystemUserTokenProvider = mockk()
+    val oppgaveBehandlingService: RestOppgaveBehandlingService = RestOppgaveBehandlingServiceImpl(
+            apiClient,
+            kodeverksmapperService,
+            pdlOppslagService,
+            tilgangskontroll,
+            ansattService,
+            arbeidsfordelingService,
+            stsService
+    )
+
+    @Test
+    fun `skal opprette oppgave`() {
+        every { stsService.systemUserToken } returns "DummyToken"
+        every { apiClient.opprettOppgave(any(), any(), any()) } returns RestOppgaveMockFactory.mockOpprettOppgaveResponse.asPost()
+        every { kodeverksmapperService.mapUnderkategori(any()) } returns Optional.empty()
+        every { kodeverksmapperService.mapOppgavetype(any()) } returns "SPM_OG_SVR"
+        every { pdlOppslagService.hentIdent(any()) } returns HentIdent.Identliste(listOf(HentIdent.IdentInformasjon("07063000250", HentIdent.IdentGruppe.AKTORID)))
+
+        SubjectHandlerUtil.withIdent("Z999998") {
+            oppgaveBehandlingService.opprettOppgave(
+                    OpprettOppgaveRequest(
```

Comment:
Kan vi ikk slå samme `OprettOppgaveRequest` og `PostOppgaveRequestJsonDTO`

Reply:
kanskje vi kan bruke builder pattern for hente inn nok informasjon som vi trenger for OprettOppgaveRequest ellers andre verdier kunne være null ellers " " som vi trenger ikke å spesifisere